bookmark-it/scripts/update-icon.js

```diff
@@ -0,0 +1,91 @@
+/**
+In this file:
+
+Update the browser action's icon when:
+* the user changes the active tab (onActivated)
+* the user loads a new page into the active tab (onUpdated)
+* the add-on is installed or enabled.
+*/
+
+/**
+Values for the icon when the page *is not* bookmarked.
+*/
+const notBookmarked = {
+  19: "icons/star-empty-19.png",
+  38: "icons/star-empty-38.png"
+};
+
+/**
+Values for the icon when the page *is* bookmarked.
+*/
+const bookmarked = {
+  19: "icons/star-filled-19.png",
+  38: "icons/star-filled-38.png"
+}
+
+
+/**
+If the array is empty, the page is not bookmarked.
+Otherwise, the page is bookmarked.
+*/
+function updateIcon(bookmarkArray) {
+  if (bookmarkArray.length === 0) {
+    browser.browserAction.setIcon({
```

Comment:
Similarly, this promise should be returned so your call below can catch errors:
````js
return browser.browserAction.setIcon({...});
````